cmd/gdeploy/main.go

```diff
@@ -268,3 +270,36 @@ func PreventDevUsage() cli.BeforeFunc {
 		return nil
 	}
 }
+
+// Check whether gdeploy version matches with the granted-approval.
+func VerifyGDeployCompatibility() cli.BeforeFunc {
+	return func(c *cli.Context) error {
+		ctx := c.Context
+		dc, err := deploy.ConfigFromContext(ctx)
+		if err != nil {
+			return err
+		}
+
+		// skip compatibility check for dev deployments.
+		if dc.Deployment.Dev != nil && *dc.Deployment.Dev {
+			return nil
+		}
+
+		isValidReleaseNumber, err := regexp.MatchString(`v\d.\d+.\d+`, dc.Deployment.Release)
+		if err != nil {
+			return err
+		}
+
+		if isValidReleaseNumber && build.Version != dc.Deployment.Release {
+			return clio.NewCLIError("Uncompatible gdeploy version and granted-approval version. You need to update gDeploy to the latest version.")
+		}
+
+		// release value are added as URL for UAT. In such case it should skip this check.
```

Comment:
This block needs to come above the isValidReleaseNumber check otherwise url based releases will always fail